### ISSUE TITLE
no_unneeded_control_parentheses have no effect

### ISSUE THREAD

**Issue by nazarhussain** [FriendsOfPHP/PHP-CS-Fixer#2798]
PHP version `5.6.30`, CS Fixer version `2.3.1` 

I used command  `./vendor/bin/php-cs-fixer fix -v --diff --dry-run <file path>`

I have set `'no_unneeded_control_parentheses' => ['break', 'clone', 'continue', 'echo_print', 'return', 'switch_case', 'yield'],` in my configuration. 

Here is the sample code: 

```
switch ($status) {
    case 2: {
       // Single line 
    }
    break;

    case 3: {
       // Multline code 
       // Multline code 
       // Multline code 
    }
    break;

}
```

CS Fixer don't detect any error here, as parentheses are not needed at least in one line code block. 

Can any one help out in it.

**Comment by SpacePossum** [comment]
Hi!
Thanks for the detailed report. Your case is known and a feature request to remove such braces has been done here https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1540 , sadly no one found the time to finish a good PR for the feature. As we would like to have it, please let me know if you've time to help out! :)

**Comment by nazarhussain** [comment]
@SpacePossum If you can guide me how to stat with it, I will try to get it done.

**Comment by SpacePossum** [comment]
sure, I started on this one time, it might be a nice base to get going;
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1883

in order to get ready:
* the fixer needs to be modernized to the new standards
* it must be able to deal with the following cases currently not covered by the tests:

```php
<?php

$c=  1;
$b = 'c';
$a = 1;

switch($a) {
    case (1 === $a ? true : ${$b}) : {
        echo 456;
        break;
    }

    case ${$b}: {{
        echo 123;
        break;
    }}
}
```

**Comment by SpacePossum** [comment]
other way could be to modify the `no_unneeded_control_parentheses` fixer, which might be easier now I think about it

**Comment by nazarhussain** [comment]
@SpacePossum I believe you reached to solution. As you have more insights to this library you can fix it quickly. That will be great help for community. Because this fix will help to improve code as per `PSR2`

**Comment by keradus** [comment]
Closed by #2929
